src/cmd/package.go

```diff
@@ -203,6 +203,17 @@ var packageRemoveCmd = &cobra.Command{
 			message.Fatalf(err, lang.CmdPackageRemoveErr, err.Error())
 		}
 	},
+	ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		var removeCandidates []string
+		// Get all the deployed packages
+		deployedZarfPackages, _ := cluster.NewClusterOrDie().GetDeployedZarfPackages()
+		// Populate list of package names
+		for _, pkg := range deployedZarfPackages {
+			removeCandidates = append(removeCandidates, pkg.Name)
+		}
+
+		return removeCandidates, cobra.ShellCompDirectiveDefault
+	},
```

Comment:
```suggestion
	ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
		var removeCandidates []string

		c, err := cluster.NewCluster()
		if err != nil {
			return removeCandidates, cobra.ShellCompDirectiveDefault
		}

		// Get all the deployed packages
		deployedZarfPackages, _ := c.GetDeployedZarfPackages()
		// Populate list of package names
		for _, pkg := range deployedZarfPackages {
			removeCandidates = append(removeCandidates, pkg.Name)
		}

		return removeCandidates, cobra.ShellCompDirectiveDefault
	},
```

Reply:
Small change to prevent a delay when no cluster exists - `NewClusterOrDie` will wait up to 30 seconds for a cluster connection which causes an apparent freeze on the CLI until that completes - there are some Zarf packages (like those that deploy cluster pre-reqs) that _might_ not have a cluster during remove.